Source/WebCore/rendering/RenderThemeIOS.mm

```diff
@@ -1328,21 +1328,10 @@ static bool shouldUseConvexGradient(const Color& backgroundColor)
         thumbnailPictureFrameRect.contract(kMultipleThumbnailShrinkSize, kMultipleThumbnailShrinkSize);
         thumbnailRect.contract(kMultipleThumbnailShrinkSize, kMultipleThumbnailShrinkSize);
 
-        // Background picture frame and simple background icon with a gradient matching the button.
+        // Background picture frame and simple background icon.
         Color backgroundImageColor = buttonRenderer.style().visitedDependentColor(CSSPropertyBackgroundColor);
         paintInfo.context().fillRoundedRect(FloatRoundedRect(thumbnailPictureFrameRect, cornerSize, cornerSize, cornerSize, cornerSize), pictureFrameColor);
         paintInfo.context().fillRect(thumbnailRect, backgroundImageColor);
-        {
-            GraphicsContextStateSaver stateSaver2(paintInfo.context());
-            CGContextRef cgContext = paintInfo.context().platformContext();
-            paintInfo.context().clip(thumbnailRect);
-            if (shouldUseConvexGradient(backgroundImageColor))
-                drawAxialGradient(cgContext, gradientWithName(ConvexGradient), thumbnailRect.location(), FloatPoint(thumbnailRect.x(), thumbnailRect.maxY()), LinearInterpolation);
```

Comment:
Are there any remaining clients of ConvexGradient, ShadeGradient, or ShineGradient, or can we follow this deletion to those?

Reply:
yes, for example all are used within `paintPushButtonDecorations`

Reply:
That code can probably be removed now too... but definitely a separate PR, and unrelated to the work you've been doing.